### PR TITLE
Improve import system

### PR DESCRIPTION
**Pull request by jonbesga**
# Description

> **Note**
>
> Please provide a description of the work completed in this PR below
>
>

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Documentation update
- [x] Project configuration change

## Complexity

> **Note**
>
> Please provide an estimated complexity of this PR of either Low, Medium or High
>
>

Complexity:

## How Has This Been Tested?
- [ ] Unit tests
- [ ] E2E tests
- [ ] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [ ] Unit tests have been updated
- [ ] E2E tests have been updated
- [ ] Examples have been updated
- [ ] New files have copyright header
- [ ] CLA (https://neo4j.com/developer/cla/) has been signed
- [ ] CHANGELOG.md updated if appropriate

### CODE REVIEW

File: tests/unit/llm/test_anthropic_llm.py [hunk 1]
```diff
@@ -17,14 +17,18 @@
...
 
 
-def get_mock_anthropic() -> MagicMock:
+@pytest.fixture
+def mock_anthropic() -> Generator[MagicMock, None, None]:
     mock = MagicMock()
     mock.APIError = anthropic.APIError
-    return mock
+
+    with patch.dict(sys.modules, {"anthropic": mock}):
```
Comment (stellasia) [inline tests/unit/llm/test_anthropic_llm.py:30]: If I understand correctly, this is overwriting `sys.modules` right? Are we sure we're not deleting anything important like another module?

**Review by willtai — COMMENTED** [review]
Nice one! What do you think of doing the same for the external retrievers: Pinecone, Weaviate, and Qdrant? If not we can do this in a separate PR

**Comment by jonbesga** [comment]
> Nice one! What do you think of doing the same for the external retrievers: Pinecone, Weaviate, and Qdrant? If not we can do this in a separate PR

Yep. I'll do the same for everything. But Qdrant has a PR open so maybe leave that for when that PR is merged?

**Comment by jonbesga** [comment]
> Nice one! What do you think of doing the same for the external retrievers: Pinecone, Weaviate, and Qdrant? If not we can do this in a separate PR

Actually, I'll do it in a separate PR

File: pyproject.toml [hunk 2]
```diff
@@ -31,63 +31,49 @@ from = "src"
...
 [tool.poetry.extras]
-external_clients = ["weaviate-client", "pinecone-client", "google-cloud-aiplatform", "cohere", "anthropic", "mistralai", "qdrant-client"]
+weaviate = ["weaviate-client"]
+pinecone = ["pinecone-client"]
+google = ["google-cloud-aiplatform"]
+cohere = ["cohere"]
+anthropic = ["anthropic"]
+openai = ["openai"]
+mistralai = ["mistralai"]
+llama-index = ["llama-index"]
```
Comment (stellasia) [inline pyproject.toml:72]: Do we also need a langchain group? We have a wrapper for langchain text splitters as well.

Comment (jonbesga) [reply pyproject.toml:72]: I can only see that we use it in experimental?

Comment (stellasia) [reply pyproject.toml:72]: Ok, you're right, `langchain-text-splitters` is in the experimental group and that's enough. 

Other question then: shouldn't be the same for llama-index?

Comment (jonbesga) [reply pyproject.toml]: You are right. Fixed c554143035e5a9a30262d33c0526a40d0fcf8af6

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,11 @@
+[tox]
```
Comment (stellasia) [inline tox.ini:1]: My total ignorance here: what is tox used for?

Comment (jonbesga) [reply tox.ini:1]: Multi-version testing!

**Review by stellasia — APPROVED** [review]
👩🏽‍🚒

**Pull request merged**